src/lib.rs

```diff
@@ -117,11 +117,23 @@ fn ulid_to_timestamp(input: ulid) -> Timestamp {
     Timestamp::try_from(inner * 1000).unwrap()
 }
 
+#[pg_extern(immutable, parallel_safe)]
+fn timestamp_to_ulid(input: Timestamp) -> ulid {
+    let epoch: f64 = input.extract_part(DateTimeParts::Epoch).unwrap().try_into().unwrap();
```

Comment:
Don't you need to modify the timestamp by 946684800000?

Reply:
Actually no, I'm really curious to know why in the other way around it's needed. But all tests pass without it, tested in a database with 1.000.000 rows

Reply:
Do you have a doc link for `.extract_part(DateTimeParts::Epoch)` function? I think that might be doing this.

Reply:
I think it just calls the Postgres extract function. [Docs here](https://docs.rs/pgrx/latest/src/pgrx/datum/datetime_support/mod.rs.html#191-208)

Reply:
Yeah, Postgres modify the value by itself. [Link to an older version of the postgres docs](https://www.postgresql.org/docs/8.1/functions-datetime.html#:~:text=SELECT%20TIMESTAMP%20WITH%20TIME%20ZONE%20%27epoch%27%20%2B%20982384720%20*%20INTERVAL%20%271%20second%27%3B)

Reply:
The closest I could find is this part of the old docs from Postgres that states that the [timestamp is stored as seconds from 2000-01-01](https://www.postgresql.org/docs/9.1/datatype-datetime.html#:~:text=or%20after%20midnight-,2000%2D01%2D01,-.%20When%20timestamp%20values). But if you extract the epoch part directly it gets from 1970-01-01

Reply:
The actual way of creating a timestamp from the epoch would be calling `to_timestamp()` function on ppstgres, but I don't know if pgrx supports it